hnswlib/bruteforce.h

```diff
@@ -5,163 +5,176 @@
-                dict_external_to_internal[label] = idx;
-                cur_element_count++;
             }
+            memcpy(data_ + size_per_element_ * idx + data_size_, &label, sizeof(labeltype));
+            memcpy(data_ + size_per_element_ * idx, datapoint, data_size_);
         }
-        memcpy(data_ + size_per_element_ * idx + data_size_, &label, sizeof(labeltype));
-        memcpy(data_ + size_per_element_ * idx, datapoint, data_size_);
-    }
-
-
-    void removePoint(labeltype cur_external) {
-        size_t cur_c = dict_external_to_internal[cur_external];
-
-        dict_external_to_internal.erase(cur_external);
-
-        labeltype label = *((labeltype*)(data_ + size_per_element_ * (cur_element_count-1) + data_size_));
-        dict_external_to_internal[label] = cur_c;
-        memcpy(data_ + size_per_element_ * cur_c,
-                data_ + size_per_element_ * (cur_element_count-1),
-                data_size_+sizeof(labeltype));
-        cur_element_count--;
-    }
-
-
-    std::priority_queue<std::pair<dist_t, labeltype >>
-    searchKnn(const void *query_data, size_t k, BaseFilterFunctor* isIdAllowed = nullptr) const {
-        assert(k <= cur_element_count);
-        std::priority_queue<std::pair<dist_t, labeltype >> topResults;
-        if (cur_element_count == 0) return topResults;
-        for (int i = 0; i < k; i++) {
-            dist_t dist = fstdistfunc_(query_data, data_ + size_per_element_ * i, dist_func_param_);
-            labeltype label = *((labeltype*) (data_ + size_per_element_ * i + data_size_));
-            if ((!isIdAllowed) || (*isIdAllowed)(label)) {
-                topResults.push(std::pair<dist_t, labeltype>(dist, label));
-            }
+
+        void removePoint(labeltype cur_external)
+        {
+            size_t cur_c = dict_external_to_internal[cur_external];
+
+            dict_external_to_internal.erase(cur_external);
+
+            labeltype label = *((labeltype *)(data_ + size_per_element_ * (cur_element_count - 1) + data_size_));
+            dict_external_to_internal[label] = cur_c;
+            memcpy(data_ + size_per_element_ * cur_c,
+                   data_ + size_per_element_ * (cur_element_count - 1),
+                   data_size_ + sizeof(labeltype));
+            cur_element_count--;
         }
-        dist_t lastdist = topResults.empty() ? std::numeric_limits<dist_t>::max() : topResults.top().first;
-        for (int i = k; i < cur_element_count; i++) {
-            dist_t dist = fstdistfunc_(query_data, data_ + size_per_element_ * i, dist_func_param_);
-            if (dist <= lastdist) {
-                labeltype label = *((labeltype *) (data_ + size_per_element_ * i + data_size_));
-                if ((!isIdAllowed) || (*isIdAllowed)(label)) {
+
+        std::priority_queue<std::pair<dist_t, labeltype>>
+        searchKnn(const void *query_data, size_t k, BaseFilterFunctor *isIdAllowed = nullptr, const std::optional<size_t> ef_search = std::nullopt) const
```

Comment:
Argument exists here to conform to the interface, but actually does nothing in bruteforce.